pynamodb/models.py

```diff
@@ -324,7 +324,7 @@ def batch_get(
         :param items: Should be a list of hash keys to retrieve, or a list of
             tuples if range keys are used.
         """
-        items = list(items)
+        items = list(set(items))
```

Comment:
All we're doing with `items` is just `while items:` (== `items.__len__`) and `items.pop()`, both are supported on a set:
```suggestion
        items = set(items)
```